### ISSUE TITLE
Disabled components should still receive focus

### ISSUE THREAD

**Issue by dzearing** [OfficeDev/office-ui-fabric-react#4387]
Disabled components should be focusable so that blind users can tab through things which as disabled and know they are disabled.

All components which render a disabled state would be affected.

List of components which can be disabled:

- [X] Buttons (using `allowDisabledFocus`)
- [ ] Checkbox
- [ ] ChoiceGroup
- [X] ComboBox
- [ ] ContextualMenu items
- [ ] DatePicker
- [ ] Dropdown
- [ ] Pickers
- [ ] Link
- [ ] Rating
- [ ] SearchBox
- [ ] Slider
- [X] SpinButton
- [ ] SwatchColorPicker
- [ ] TextField
- [ ] Toggle

Codepen which renders all of them:

https://codepen.io/micahgodbolt/pen/VqzERG

**Comment by michaelangotti** [comment]
@betrue-final-final Tagging you since this is Accessibility.

**Comment by dzearing** [comment]
This just needs a dev to make it work, I believe.

@betrue-final-final Right? We're in agreement here?

I think there will be some edge cases. For example, if you have a dropdown with disabled options, should those be focusable?

What about a pivot tab or disabled ChoiceGroup option.

**Comment by micahgodbolt** [comment]
hasn't there been work done on this? Is there a more holistic approach we need to take?

**Comment by betrue-final-final** [comment]
I need to set up an actual meeting again to see pull in other stakeholders to make sure were doing the right thing. I thought we had closure but it keeps coming back.

**Comment by micahgodbolt** [comment]
@betrue-final-final is this still worth doing? Did you ever hear back from Jeff?

**Comment by dzearing** [comment]
I am going to close this. There isn't a single framework out there I have found which makes disabled components focusable.

Material UI does not: https://material-ui.com/components/buttons/
Carbon does not: https://www.carbondesignsystem.com/components/button/code
Ant design does not: https://ant.design/components/button/#components-button-demo-disabled
Base ui does not: https://baseweb.design/components/button/
Blueprint does not: https://blueprintjs.com/docs/#core/components/button

The browser itself prevents focus, through tabindex, or even imperative `.focus` calls to set focus on disabled items. We are fighting against the platform here, which isn't what users expect. Closing.
